src/store/cache/useOutputsCacheStore.ts

```diff
@@ -52,20 +52,24 @@ const createOutputsStore: StateCreator<OutputsStore> = (set, get) => ({
 				outputs = [...currentSession.outputs, ...logs];
 			}
 
-			const isSessionFinished =
-				[SessionStateType.error, SessionStateType.completed, SessionStateType.stopped].includes(
-					sessionStateConverter(sessionInfo.state) as SessionStateType
-				) && !nextPageToken;
+			const isSessionFinished = [
+				SessionStateType.error,
+				SessionStateType.completed,
+				SessionStateType.stopped,
+			].includes(sessionStateConverter(sessionInfo.state) as SessionStateType);
 
-			if (!currentSession.hasLastSessionState) {
+			if (!currentSession.hasLastSessionState && isSessionFinished) {
```

Comment:
[On line 48](https://github.com/autokitteh/web-platform/blob/b31b763d625352ca0c88da187b58c9ed8063fe7a/src/store/cache/useOutputsCacheStore.ts#L48): `let outputs;`
You can use c`onst outputs = force ? [...logs] : [...currentSession.outputs, ...logs];` since you're not reassigning the variable itself.